### PR TITLE
[NFC] Rename tensor properties.

### PR DESCRIPTION
**Pull request by dan-zheng**
Use established terms-of-art for tensor properties.

I don't mean to get caught up in bikeshedding, but I believe the established terms are a bit easier to understand and more usable.

* `dims` -> `shape`
* `nbDims` -> `rank`
* `nbElem` -> `scalarCount`

### CODE REVIEW

**Comment by dan-zheng** [comment]
@feiwang3311 I'm curious about the purpose/meaning of `def dims = TTT(dimsSeq)` (I'm using the old variable names).

What does `TTT` accomplish? How should I decide whether to use `dimsSeq` or `dims`?

**Comment by feiwang3311** [comment]
Thanks, Dan. That is good suggestion. As to TTT, Greg put it there and it doesn't seem to provide anything more than dimsSeq. In most cases I just use dims.

**Comment by GSAir** [comment]
The TTT was added as a debugging tool. It was to be able to intercept call to apply in the Seq in order to do range checking. For some reason I could get the StackTrace otherwise if I recall correctly. It can be removed. Or we should create a proper object that take an input of dimensions and have some function that can help with the strides and direct access in the Array.

**Comment by TiarkRompf** [comment]
I agree with the renaming. It's good to use standard terminology.

File: src/main/scala/lantern/ad_lms_vector.scala [hunk 1]
```diff
@@ -256,199 +254,198 @@ trait TensorExp extends Dsl with Diff {
   class BackendDefault extends BackendNative
   val backend: Backend = new BackendDefault
 
-  class Tensor(val data: Rep[Array[Float]], val dimsSeq: NSeq[Int]) extends Serializable {
+  class Tensor(val data: Rep[Array[Float]], val dimensions: NSeq[Int]) extends Serializable {
```
Comment (TiarkRompf) [inline src/main/scala/lantern/ad_lms_vector.scala]: Should this just be called `shape`?

Comment (dan-zheng) [reply src/main/scala/lantern/ad_lms_vector.scala]: [There's already something called `shape`](https://github.com/feiwang3311/Lantern/pull/13/files#diff-731c21d511864089e5fa75e2e69d72aaR266), so I named this guy `dimensions`.

The current `shape` (i.e. `TTT(dimensions)`) is meant to be used everywhere. It enables debugging, as @GSAir explains above.

**Pull request merged**